src/machine_learning/api.py

```diff
@@ -112,5 +106,56 @@ def add_controlnet_segmentation(self, model: str, seg_map: str, depth_map: str =
             ],
         }
 
-    def to_dict(self) -> dict:
-        return self.config
+class Img2ImgConfig(X2ImgConfig):
+    """
+    Config for img2img API
+    """
+    def __init__(self, **kwargs) -> None:
+        """Initialize the config with default values. Use kwargs to override the defaults."""
+        super().__init__(**kwargs)
+        self.config.update({
+            "init_images": [],
+            "resize_mode": 0,
+            "denoising_strength": 0.75,
+            "image_cfg_scale": 0,
+            "override_settings": {},
+            "override_settings_restore_afterwards": True,
+            "include_init_images": False,
+            "send_images": True,
+            "save_images": False,
+        })
+        self.config.update(kwargs)
+
+    def set_init_images(self, images: list):
+        """
+        Set the initial images for the config.
+
+        Args:
+            images (list): A list of base64-encoded images.
+
+        Returns:
+            None
+        """
+        self.config["init_images"] = images
+
+class Txt2ImgConfig(X2ImgConfig):
```

Comment:
Same here, a different file for a class is preferred.

Reply:
> Same here, a different file for a class is preferred.

I will separate the StableDiffusionAPI class from the other `XXX2ImgConfig` Class.

The `X2ImgConfig` and its subclasses have a close relationship with each other. The resulting file would have an acceptable length.